### ISSUE TITLE
sheared feature mapping

### ISSUE THREAD

**Issue by Luke199010** [OpenDriveLab/PersFormer_3DLane#117]
Hello, 
     Why is sheared_feature_map obtained by sampling on x_feature_map?
I don't understand what sheared_feature_map does? This sampling method is a bit like sampling lane line features in LaneATT. The difference is that you fill in 0 in other positions of the feature map, and only sample on straight lines with specific angles, and these specific angles do not seem to be consistent with the angle of the anchor.

            v_arange = torch.arange(fmap_h).unsqueeze(dim=1).repeat(1,fmap_w*6).to(x.device)
            self.fmap_mapping_interp_index = self.fmap_mapping_interp_index.to(x.device)
            self.fmap_mapping_interp_weight = self.fmap_mapping_interp_weight.to(x.device)

            for batch_idx, x_feature_map in enumerate(x):
                # if True:
                # print("v_arange device: " + str(v_arange.device))
                # print("self.fmap_mapping_interp_index device: " + str(self.fmap_mapping_interp_index.device))
                # print("self.fmap_mapping_interp_weight device: " + str(self.fmap_mapping_interp_weight.device))
                # print("sheared_feature_map device: " + str(sheared_feature_map.device))
                # print("batch_idx device: " + str(v_arange.device))
                # print("x_feature_map device: " + str(x_feature_map.device))

                sheared_feature_map[batch_idx] = \
                    x_feature_map[:, v_arange, self.fmap_mapping_interp_index[:,:,0]] * self.fmap_mapping_interp_weight[:,:,0] + \
                    x_feature_map[:, v_arange, self.fmap_mapping_interp_index[:,:,1]] * self.fmap_mapping_interp_weight[:,:,1]
            x = torch.cat((x, sheared_feature_map), dim=3)

Looking forward to your reply. Thank you.
